app/src/main/java/dev/leonlatsch/photok/gallery/ui/GalleryFragment.kt

```diff
@@ -143,11 +143,24 @@ class GalleryFragment : BindableFragment<FragmentGalleryBinding>(R.layout.fragme
     }
 
     private val onAdapterDataObserver = object : RecyclerView.AdapterDataObserver() {
-        override fun onItemRangeInserted(positionStart: Int, itemCount: Int) =
+        override fun onItemRangeInserted(positionStart: Int, itemCount: Int) {
             viewModel.togglePlaceholder(adapter.itemCount)
+            updateFileAmount()
+        }
 
-        override fun onItemRangeRemoved(positionStart: Int, itemCount: Int) =
+        override fun onItemRangeRemoved(positionStart: Int, itemCount: Int)  {
             viewModel.togglePlaceholder(adapter.itemCount)
+            updateFileAmount()
+        }
+    }
+
+    private fun updateFileAmount() {
+        if (adapter.itemCount == 0) {
+            binding.galleryAllPhotosAmount.visibility = View.GONE
+        } else {
+            binding.galleryAllPhotosAmount.visibility = View.VISIBLE
+            binding.galleryAllPhotosAmount.text = "(${adapter.itemCount})"
```

Comment:
This way, we only show the number of files that are currently visible, since they are loaded in pages. Just to have it in mind, I don't know if its the right way

Reply:
@leonlatsch Seems to be working for me pretty well. I just imported over 500 files into the app and it shows an exact number. After a restart I immediately get that number again.